File: src/common/net/auoeke/wheel/loader/DelegateTransformingLoader.java
package net.auoeke.wheel.loader;

import java.util.Arrays;

public class DelegateTransformingLoader extends TransformingClassLoader {
    public DelegateTransformingLoader(TransformingClassLoader parent) {
        super(parent.getName() + "-delegate", parent, Arrays.asList(parent.getURLs()));
    }

    @Override
    public Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundException {
        return ((TransformingClassLoader) this.getParent()).loadClass(name, resolve);
    }
}

File: src/net/auoeke/wheel/loader/DelegateTransformingLoader.kt
package net.auoeke.wheel.loader

class DelegateTransformingLoader(parent: TransformingClassLoader) : TransformingClassLoader(parent.name + "-delegate", parent, listOf(*parent.urLs)) {
    override fun loadClass(name: String, resolve: Boolean): Class<*> {
        return (this.parent as TransformingClassLoader).loadClass(name, resolve)
    }
}
